app/src/main/java/com/android/sample/ui/createRecipe/CategoryScreen.kt

```diff
@@ -0,0 +1,171 @@
+ * @param navigationActions Actions for navigating between screens.
+ * @param createRecipeViewModel ViewModel for managing the recipe creation process.
+ */
+@Composable
+fun CategoryContent(
+    modifier: Modifier = Modifier,
+    currentStep: Int,
+    navigationActions: NavigationActions,
+    createRecipeViewModel: CreateRecipeViewModel
+) {
+  val noCategoryString = stringResource(R.string.no_category)
+  val categories = listOf(noCategoryString) + Recipe.getCategories()
+  val selectedCategory = remember { mutableStateOf(createRecipeViewModel.getRecipeCategory()) }
+  val expanded = remember { mutableStateOf(false) }
+
+  Box(modifier = modifier.padding(PADDING_8.dp), contentAlignment = Alignment.TopCenter) {
+    Column(
+        modifier = Modifier.fillMaxSize(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Top) {
+          // Display the progress bar for the current step
+          RecipeProgressBar(currentStep = currentStep)
+
+          Spacer(modifier = Modifier.weight(1f))
+
+          // Title
+          Text(
+              text = stringResource(R.string.select_category),
+              style = Typography.displayLarge,
+              color = MaterialTheme.colorScheme.onPrimary,
+              modifier =
+                  Modifier.fillMaxWidth()
+                      .padding(horizontal = PADDING_16.dp)
+                      .testTag("CategoryTitle"),
+              textAlign = TextAlign.Center)
+
+          Spacer(modifier = Modifier.weight(1f))
+
+          // Subtitle
+          Text(
+              text = stringResource(R.string.select_category_description_optional),
+              style = MaterialTheme.typography.bodyMedium,
+              color = MaterialTheme.colorScheme.onPrimary,
+              modifier = Modifier.padding(horizontal = PADDING_32.dp).testTag("CategorySubtitle"),
+              textAlign = TextAlign.Center)
+
+          Spacer(modifier = Modifier.weight(2f))
+
+          // Dropdown menu for selecting category
+          Box(modifier = Modifier.fillMaxWidth().padding(horizontal = PADDING_16.dp)) {
+            OutlinedButton(
+                onClick = { expanded.value = !expanded.value },
+                modifier = Modifier.fillMaxWidth().testTag(DROPDOWN_TEST_TAG),
+                shape = RoundedCornerShape(DROPDOWN_CORNER_RADIUS.dp),
+                colors =
+                    ButtonDefaults.outlinedButtonColors(
+                        containerColor = MaterialTheme.colorScheme.surface,
+                        contentColor = MaterialTheme.colorScheme.onSurface)) {
+                  Text(
+                      text =
+                          selectedCategory.value ?: stringResource(R.string.no_category_selected),
+                      style = MaterialTheme.typography.bodyMedium,
+                      modifier = Modifier.padding(vertical = PADDING_8.dp))
+                }
+
+            DropdownMenu(
```

Comment:
I noticed a padding issue with the `DropdownMenu`. In the attached picture, the padding is uneven on the left and right sides. I believe this is caused by the `DropdownMenu` modifier using `fillMaxWidth` combined with a fractional height. It would be better to replace this approach with explicit padding to ensure consistent spacing.

<img src="https://github.com/user-attachments/assets/57cef91e-8e7c-4b8e-9fea-907cdf504d42" width="350" />